lib/SlackEventHandler.js

```diff
@@ -178,42 +211,37 @@ SlackEventHandler.prototype.handleMessageEvent = function (params) {
         // just send the message as text.
         log.warn("no slack token for " + room.getSlackTeamDomain() || room.getSlackChannelId());
 
-        return (processMsg) ? room.onSlackMessage(msg) : Promise.resolve();
+        if (processMsg) {
+            await room.onSlackMessage(msg);
+        }
+        return;
     }
 
     if (!processMsg) {
         // TODO(paul): When I started looking at this code there was no lookupAndSendMessage()
         //   I wonder if this code path never gets called...?
         // lookupAndSendMessage(params.channel_id, params.timestamp, intent, roomID, token);
         log.warn("Did not understand event: ", JSON.stringify(message));
-        return Promise.resolve();
+        return;
     }
 
-    var result;
     if (msg.subtype === "file_share" && msg.file) {
-        // TODO check is_public when matrix supports authenticated media https://github.com/matrix-org/matrix-doc/issues/701
+        // TODO check is_public when matrix supports authenticated
+        // media https://github.com/matrix-org/matrix-doc/issues/701
         // we need a user token to be able to enablePublicSharing
         if (room.getSlackUserToken()) {
-            result = this.enablePublicSharing(msg.file, room.getSlackUserToken())
-                .then((file) => {
-                    if (file) {
-                        msg.file = file;
-                    }
-
-                    return this.fetchFileContent(msg.file, token)
-                        .then((content) => {
-                            msg.file._content = content;
-                        });
-                })
+            const file = this.enablePublicSharing(msg.file, room.getSlackUserToken());
+            if (file) {
+                msg.file = file;
+                const content = await this.fetchFileContent(msg.file, token);
+                msg.file._content = content;
+            }
         }
-    } else {
-        result = Promise.resolve();
     }
 
-    return result.then(() => msg)
```

Comment:
Be careful here that the original code is returning the result of `onSlackMessage` to the caller, wheras you are returning `undefined`. I don't know if it's an issue.